pages/index.js

```diff
@@ -6,149 +6,116 @@ import { Typography, Col, Row } from 'antd';
     talks: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string })).isRequired,
+    requestState: PropTypes.shape({ ifLoaded: PropTypes.func }),
 };
 
-const EmptyMonthCard = ({ date }) => (
-    <Card title="Next empty month">{format(new Date(date), 'MMMM')}</Card>
+const EmptyMonthCard = ({ date, requestState }) => (
+    <Card title="Next empty month" requestState={requestState}>
+        {requestState.ifLoaded(() => format(new Date(date), 'MMMM'))}
+    </Card>
 );
 
 EmptyMonthCard.propTypes = {
     date: PropTypes.string.isRequired,
+    requestState: PropTypes.shape({ ifLoaded: PropTypes.func }),
 };
 
 const Talks = () => {
-    const {
-        data: { nextTalk, plannedTalks, nextEmptyMonth },
-        isLoading,
-        isError,
-        retry,
-    } = useDataApi('/api/talks');
-
-    if (isError) {
-        return (
-            <React.Fragment>
-                <CardWithError title="Next talk" onFailureRetry={retry} />
-                <CardWithError title="Planned talks" onFailureRetry={retry} />
-                <CardWithError title="Next empty month" onFailureRetry={retry} />
-            </React.Fragment>
-        );
-    }
-
-    if (isLoading) {
-        return (
-            <React.Fragment>
-                <Col>
-                    <CardWithLoading title="Next talk" />
-                </Col>
-                <Col>
-                    <CardWithLoading title="Planned talks" />
-                </Col>
-                <Col>
-                    <CardWithLoading title="Next empty month" />
-                </Col>
-            </React.Fragment>
-        );
-    }
+    const requestState = useDataApi('/api/talks');
+    const { nextTalk, plannedTalks, nextEmptyMonth } = requestState.data;
 
     return (
         <React.Fragment>
             <Col>
-                <NextTalkCard {...nextTalk} />
+                <NextTalkCard {...nextTalk} requestState={requestState} />
             </Col>
             <Col>
-                <PlannedTalksCard talks={plannedTalks} />
+                <PlannedTalksCard talks={plannedTalks} requestState={requestState} />
             </Col>
             <Col>
-                <EmptyMonthCard date={nextEmptyMonth} />
+                <EmptyMonthCard date={nextEmptyMonth} requestState={requestState} />
             </Col>
         </React.Fragment>
     );
 };
 
 const NextTweet = () => {
-    const {
-        data: { nextTweet },
-        isLoading,
-        isError,
-        retry,
-    } = useDataApi('/api/communication');
-
-    if (isError) return <CardWithError title="Next tweet" onFailureRetry={retry} />;
-    if (isLoading) return <CardWithLoading title="Next tweet" />;
-
-    if (!nextTweet.date) {
-        return (
-            <Card title="Next tweet" state="error">
-                <div>No tweet has been prepared!</div>
-            </Card>
-        );
-    }
+    const requestState = useDataApi('/api/communication');
+    const nextTweet = requestState.data.nextTweet;
```

Comment:
```suggestion
    const { nextTweet }  = requestState.data;
```

Bu it's ok like this :)